.github/workflows/ci-publish-release.yml

```diff
@@ -39,19 +40,35 @@ jobs:
         run: chmod +x ./gradlew
 
       - name: Build
-        run: ./gradlew --warning-mode all check build -x test
+        id: build
+        run: ./gradlew --warning-mode all check build -x test -x integrationTest
 
-      - name: Test
+      - name: Run Unit Tests
+        id: unit_test
+        if: ${{ steps.build.outcome == 'success' }}
         run: |
           export PATH="$HOME/.local/bin:$PATH"
           ./gradlew --no-daemon --max-workers=2 --warning-mode=all test --continue --stacktrace
-        if: "!contains(github.event.head_commit.message, '[i am bad and i hate unit tests]')"
 
       - name: Publish Unit Test Results
         uses: EnricoMi/publish-unit-test-result-action@v1.18
         if: always()
         with:
-          files: "**/test-results/**/*.xml"
+          check_name: "Unit Test Results"
+          files: "**/test-results/test/**/*.xml"
+
+      - name: Run Integration Tests
```

Comment:
Add a separate task for running integration tests, only triggers if unit tests succeeded